src/main/java/com/rcv/CastVoteRecord.java

```diff
@@ -50,6 +52,12 @@
   private boolean isExhausted;
   // tells us which candidate is currently receiving this CVR's vote (or fractional vote)
   private String currentRecipientOfVote = null;
+  // If CVR CDF output is enabled, we store the necessary info here: for each round, the list of
+  // candidates this ballot is counting toward (0 or 1 in a single-seat contest; 0 to n in a
+  // multi-seat contest because of fractional vote transfers), and how much of the vote each is
+  // getting. As a memory optimization, if the data is unchanged from the previous round, we don't
+  // add a new entry.
+  private Map<Integer, List<Pair<String, BigDecimal>>> cdfSnapshotData = new HashMap<>();
```

Comment:
Did you delete this comment?

> does it make sense to combine this with winnerToFractionalValue?

This is actually a good question. There's definitely overlap between the two data structures. The downside of combining them would be that the new structure would contain a fair amount more data than what winnerToFractionalValue currently holds. If we expect that normal operation will often not enable the CDF output, it's better to keep these two separate. If, on the other hand, we expected CDF output to be enabled usually or always, then yes, we would want to combine them.

Reply:
Yeah, and that's why I deleted it.  I do not expect anyone to be using CDF tabulation for quite some time.